### PR TITLE
Introducing different types of log msgs

### PR DESCRIPTION
**Pull request by Basit-qc**

### CODE REVIEW

File: src/codesyncd/utils.ts [hunk 1]
```diff
@@ -205,7 +205,7 @@ export class statusBarMsgs {
 		} catch (e) {
 			// eslint-disable-next-line @typescript-eslint/ban-ts-comment
 			// @ts-ignore
-			putLogEvent(e.stack);
+			CodeSyncLogger.error(e.stack);
```
Comment (saleem-latif) [inline src/codesyncd/utils.ts:208]: Can we add a description of the error message here? to make this error more readable in the logs.

File: src/extension.ts [hunk 2]
```diff
@@ -65,7 +65,7 @@ export async function activate(context: vscode.ExtensionContext) {
 			} catch (e) {
 				// eslint-disable-next-line @typescript-eslint/ban-ts-comment
 				// @ts-ignore
-				putLogEvent(e.stack);
+				CodeSyncLogger.error(e.stack);
```
Comment (saleem-latif) [inline src/extension.ts:68]: Can we add a message here as well along side the error stack?

Comment (saleem-latif) [reply src/extension.ts:68]: Same for other occurrences as well, I am skipping the comments for others.

File: src/logger.ts [hunk 3]
```diff
@@ -8,25 +8,43 @@ import { PutLogEventsRequest } from 'aws-sdk/clients/cloudwatchlogs';
...
+	INFO: "INFO",
+	DEBUG: "DEBUG"
+};
+
+export class CodeSyncLogger {
+	static info (msg: string, userEmail?: string, additionalMsg="", retryCount=0) {
+		putLogEvent(msg, logErrorMsgTypes.INFO, userEmail, additionalMsg, retryCount);
+	}
+
+	static error (msg: string, userEmail?: string, additionalMsg="", retryCount=0) {
```
Comment (saleem-latif) [inline src/logger.ts:35]: Can you add a `critical` message type as well for errors that inhibit the successful operation of plugin and are blocking for users.

Comment (saleem-latif) [reply src/logger.ts:35]: The error types that I used with my understanding are.

`DEBUG`: for developer oriented messages, only usable for brief testing of new features and should be removed once new features are fully tested.
`INFO`: Informational messages, could be useful while debugging but can be ignore during normal execution.
`WARNING`: Mild errors that do not affect the user but should be fixed in some time frame
`ERROR`: Errors that cause a bad UX and should be fixed soon.
`CRITICAL`: Errors that are blocking for the normal operation of the plugin and should be fixed immediately.

I would like your input on the above, once these are finalised I will add these in the documentation as well.

Also, I have used error types in all uppercase.

**Review by saleem-latif — APPROVED** [review]
A few minor comments, other than that looks good 👍

File: src/extension.ts (hunk 2 above)
Comment (Basit-qc) [reply src/extension.ts:68]: Good idea, will do. Thanks

File: src/logger.ts (hunk 3 above)
Comment (Basit-qc) [reply src/logger.ts:35]: 👍 I agree. Let me add critical type as well. 
I am already using error types in upper case [here](https://github.com/codesyncapp/codesync-vs/pull/22/files#diff-7bc1140e724e284ae22084e1172568baa8d151063d02c75ed1d8805fa7796acaR22) if that's what you meant

**Review by saleem-latif — APPROVED** [review]

**Pull request merged**